### PR TITLE
add support for gamma_delta

### PR DESCRIPTION
**Pull request by nmmsv10x**

### CODE REVIEW

File: vdj_ann/src/annotate.rs [hunk 1]
```diff
@@ -2716,7 +2716,7 @@ impl ContigAnnotation {
         let mut ann = Vec::<(i32, i32, i32, i32, i32)>::new();
         annotate_seq(b, refdata, &mut ann, true, false, true);
         let mut log = Vec::<u8>::new();
-        let productive = is_valid(b, refdata, &ann, false, &mut log);
+        let productive = is_valid(b, refdata, &ann, false, &mut log, None);
```
Comment (sreenathkrishnan) [inline vdj_ann/src/annotate.rs:2719]: Add this as an input parameter

File: vdj_ann/src/transcript.rs [hunk 2]
```diff
@@ -22,7 +22,13 @@ pub fn is_valid(
     ann: &[(i32, i32, i32, i32, i32)],
     logme: bool,
     log: &mut Vec<u8>,
+    is_gd: Option<bool>,
 ) -> bool {
+    // Unwrap gamma/delta mode flag
+    let gd_mode = match is_gd {
+        Some(x) => x,
+        None => false,
+    };
```
Comment (sreenathkrishnan) [inline vdj_ann/src/transcript.rs:31]:
```suggestion
    let gd_mode = is_gd.unwrap_or(false);
```

File: vdj_ann/src/transcript.rs [hunk 3]
```diff
@@ -45,31 +51,63 @@ pub fn is_valid(
             if rheaders[t].contains("IGH") {
                 igh = true;
             }
-            if !rheaders[t].contains("5'UTR")
-                && ((m == "A" && (rheaders[t].contains("TRAV") || rheaders[t].contains("IGHV")))
+            if gd_mode {
+                if !rheaders[t].contains("5'UTR")
```
Comment (sreenathkrishnan) [inline vdj_ann/src/transcript.rs:55]: Use a single if condition as opposed to two blocks

File: vdj_ann/src/transcript.rs [hunk 4]
```diff
@@ -45,31 +51,63 @@ pub fn is_valid(
...
+                && ((m == "A" && (rheaders[t].contains("TRAV") || rheaders[t].contains("TRGV") || rheaders[t].contains("IGHV")))
```
Comment (sreenathkrishnan) [inline vdj_ann/src/transcript.rs:56]:
```suggestion
                && ((m == "A" && (rheaders[t].contains("TRAV") || (rheaders[t].contains("TRGV") && is_gd) || rheaders[t].contains("IGHV")))
```

File: vdj_ann/src/transcript.rs [hunk 5]
```diff
@@ -201,15 +245,31 @@ pub fn junction_seq(
...
+                || rheaders[t].contains("TRBJ")
+                || rheaders[t].contains("IGLJ")
+                || rheaders[t].contains("IGKJ")
+                || rheaders[t].contains("TRGJ")
+                || rheaders[t].contains("TRDJ"))
+                && p + len == refs[t].len()
+                && l + len >= TAG as usize
+            {
+                jstops.push((l + len) as i32);
+            }
```
Comment (sreenathkrishnan) [inline vdj_ann/src/transcript.rs:272]: Use `(rheaders[t].contains("TRGJ") && is_gd) || (rheaders[t].contains("TRDJ") && is_gd)`

**Comment by nmmsv10x** [comment]
This branch doesn't seem to build with enclone due to issues with prost and petgraph. investigating.

File: master.toml [hunk 6]
```diff
@@ -45,4 +45,4 @@ strum_macros = ">=0.18.0, <0.22"
 superslice = "1.0.0"
 tables = "0.1"
 vdj_types = "0.1"
-vector_utils = "0.1"
+vector_utils = "0.1"
```
Comment (sreenathkrishnan) [inline master.toml:48]:
```suggestion
vector_utils = "0.1"

```

**Review by sreenathkrishnan — APPROVED** [review]
Let's merge this. The changes are well isolated and optional

**Pull request merged**